RxSwift/Observables/Observable+Single.swift

```diff
@@ -84,7 +84,7 @@ extension ObservableType {
     - returns: The source sequence with the side-effecting behavior applied.
     */
     // @warn_unused_result(message:"http://git.io/rxs.uo")
-    public func doOn(eventHandler: (Event<E>) throws -> Void)
+    public func doOn(_ eventHandler: (Event<E>) throws -> Void)
```

Comment:
Maybe we can keep it simple and just use Swift 3.0 naming conventions first.
My intention isn't to remove any useful functionality, just to modernize the API. 

It seemed to me that Xcode autocomplete provided overloads with all combinations of optional parameters, but not sure what happened to that functionality.